### DISCUSSION TITLE
How can I enable autocomplete on Neovim?

### DISCUSSION THREAD

**Discussion by juniorrodes** [a-h/templ#686] [Q&A]
Hello everyone, I'm trying to use the neovim lsp with templ for almost a week, but the auto complete is not working, I tried to search for a solution but didn't find any.

So here is my lsp configuration for templ

```lua
vim.filetype.add({ extension = { templ = 'templ' } })

lspconfig.templ.setup {
    on_attach = on_attach,
    capabilities = capabilities,
    filetypes = {
        'templ'
    },
    cmd = { 'templ', 'lsp', '-log', '/home/juniorrodes/templLsp.log' }
}
```

Here is a [gist](https://gist.github.com/juniorrodes/ef4bb55949ce52d7d4b328e5fed4b640) with the logs.

My neovim lsp is working fine with Go and zig, but with templ is just not showing any suggestions.

I'm putting this in a Q&A because I don't know if it's a skill issue, or a actual problem with the LSP, probalbly a skill issue.

**Comment by joerdav** [comment]
Looking at the logs it could be the fact that your file is named `filename.go.templ` that's just an initial hunch though! Try renaming to `filename.templ` and try again.

**Reply by juniorrodes** [reply]
In theory, my files are named just `filename.templ`, [here](https://github.com/juniorrodes/exchange/tree/main/internal/app/views) are the files that I'm trying to complete working.
I don't know why the logs are showing .go.templ

**Reply by joerdav** [reply]
Okay! This seems like a bug then. I'll see what I can do about reproducing it.

**Reply by juniorrodes** [reply]
if you need any additional information just ask that I will try to send it.

**Reply by a-h** [reply]
I couldn't reproduce the error with your templ files. I suspect it's Neovim config.

Typical advice on debugging that is to start minimal and build up because it's hard to reproduce entire Neovim config.
